aptos-node/src/lib.rs

```diff
@@ -599,22 +604,50 @@ pub fn setup_environment_and_start_node(
         jwk_consensus_subscriptions,
     ) = state_sync::create_event_subscription_service(&node_config, &db_rw);
 
+    let peer_senders = Arc::new(OutboundPeerConnections::new());
+
     // Set up the networks and gather the application network handles
-    let peers_and_metadata = network::create_peers_and_metadata(&node_config);
-    let (
-        network_runtimes,
-        consensus_network_interfaces,
-        dkg_network_interfaces,
-        jwk_consensus_network_interfaces,
-        mempool_network_interfaces,
-        peer_monitoring_service_network_interfaces,
-        storage_service_network_interfaces,
-    ) = network::setup_networks_and_get_interfaces(
+    let peers_and_metadata = network2::create_peers_and_metadata(&node_config);
+    let (network_runtimes, mut networks) = network2::setup_networks(
         &node_config,
         chain_id,
         peers_and_metadata.clone(),
-        &mut event_subscription_service,
-    );
+        peer_senders.clone(),
+        &mut event_subscription_service);
+
+    let mut apps = ApplicationCollector::new();
+
+    // create map to lookup NetworkContext by NetworkId
+    let mut contexts = BTreeMap::new();
+    for network in networks.iter() {
+        let network_context = network.network_context();
+        contexts.insert(network_context.network_id(), network_context);
+    }
+    let contexts = Arc::new(contexts);
+
+    let app_setup = AppSetupContext{
+        node_config: node_config.clone(),
+        peers_and_metadata: peers_and_metadata.clone(),
+        peer_senders,
+        contexts,
+    };
+
+    let peer_monitoring_service_network_interfaces = network2::peer_monitoring_network_connections(&mut apps, &app_setup);
+    let storage_service_network_interfaces = network2::storage_service_network_connections(&mut apps, &app_setup);
+    let mempool_network_interfaces = network2::mempool_network_connections(&mut apps, &app_setup);
+    let consensus_network_interfaces = network2::consensus_network_connections(&mut apps, &app_setup);
+    let dkg_network_interfaces = network2::dkg_network_connections(&mut apps, &app_setup);
+    let jwk_consensus_network_interfaces = network2::jwk_consensus_network_connections(&mut apps, &app_setup);
+    let netbench_network_interfaces = network2::netbench_network_connections(&mut apps, &app_setup);
+
+    for (protocol_id, ac) in apps.iter() {
+        info!("app_int setup {} -> {} {:?}", protocol_id.as_str(), ac.label, &ac.sender);
+    }
+    let apps = Arc::new(apps);
+    for network in networks.iter_mut() {
+        network.set_apps(apps.clone());
```

Comment:
We set all apps for all network types? Traditionally, apps are specific to network type, no? e.g., consensus only operates in the validator network, etc.